src/support/hazard.c

```diff
@@ -408,18 +411,21 @@ bool
 __wt_hazard_check_assert(WT_SESSION_IMPL *session, void *ref, bool waitfor)
 {
 	WT_HAZARD *hp;
+	WT_SESSION_IMPL *s;
 	int i;
 
+	s = NULL;
 	for (i = 0;;) {
-		if ((hp = __wt_hazard_check(session, ref)) == NULL)
+		if ((hp = __wt_hazard_check(session, ref, &s)) == NULL)
 			return (true);
 		if (!waitfor || ++i > 100)
 			break;
 		__wt_sleep(0, 10000);
 	}
 	__wt_errx(session,
-	    "hazard pointer reference to discarded object: (%p: %s, line %d)",
-	    (void *)hp->ref, hp->func, hp->line);
+	    "hazard pointer reference to discarded object: "
+	    "(%p: session %p name %s: %s, line %d)",
+	    (void *)hp->ref, (void *)s, s->name, hp->func, hp->line);
```

Comment:
I think we need to be careful about `s->name == NULL` here -- that is a pointer to string constants, so if we get any non-NULL value, I think it's fine to use it.  Some `snprintf` implementations will segault on NULL, though.